src/stream/tag.rs

```diff
@@ -273,7 +273,7 @@ impl Encoder {
 
         let mut storage = PlainStorage::new(file, location);
         let mut w = storage.writer()?;
-        self.encode(tag, &mut w)?;
+        self.encode(tag, &mut w, Some(1024))?;
```

Comment:
Hardcoding this number spooks me quite a bit, we should make some effort to de-duplicate it.

The padding is also written by the storage module which makes this an interesting challenge to solve in an correct manner, hmmm...

The storage module is also able to grow and shrink the tag padding as it desires so this number will only be correct for some cases. But the padding can only be determined after the tag has been serialized which would require the storage module to either have knowledge about the ID3 format to update the header or depend on ID3 serialization to determine the padding...

Reply:
Yeah I also agree that hardcoding is dirty, but before it was hardcoded to 2048 in the storage module.

Reply:
The 2048 is determined by a fair dice roll, but was only defined there so hardcoding a default was fine. Having 1024 here and 1023 there creates an implicit relation which is bad practice.

The original value is used as a 'preferred' padding which is used for when the new tag no longer fits within the padded area and the file needs to be rewritten. This parameter controls the size of the padding that is added. The padding size therefore is not necessarily equal to this number.